lxd/storage/drivers/driver_btrfs.go

```diff
@@ -192,6 +192,8 @@ func (d *btrfs) Create() error {
 		if tryExists(fmt.Sprintf("/dev/disk/by-uuid/%s", devUUID)) {
 			// Override the config to use the UUID.
 			d.config["source"] = devUUID
+		} else {
+			return fmt.Errorf("Timeout discovering disk by UUID: %q", d.config["source"])
```

Comment:
I'm not sure if this is correct, if the /dev/disk/by-uuid entry doesn't appear then we'll keep source as the block device path specified by the user right? Isn't that preferable than failing?

Reply:
I'm open to being persuaded though - just not sure if this should fallback or fail.

Reply:
I discovered it whilst following the reproducer steps from the bug report. There was a situation in which I ran `dd` to scrap the first few bytes of the disk to remove the FS so I could recreate pool using the same disk (haven't used `source.wipe` in this case).

It looked not right having two keys `source` and `volatile.initial_source` both pointing to `/dev/sdb` in this case.
But as you said it's the fallback and it works :) But the more I think about it maybe we should keep it for compatibility reasons as it's not really crashing anything in case of a timeout.

Reply:
Maybe with a comment explaining that its there on purpose though

Reply:
Sure, comment added.